packages/library-sync/src/commonMain/kotlin/io/realm/kotlin/mongodb/internal/AppConfigurationImpl.kt

```diff
@@ -34,11 +34,13 @@ import io.realm.kotlin.mongodb.AppConfiguration
 import io.realm.kotlin.mongodb.AppConfiguration.Companion.DEFAULT_BASE_URL
 
 // TODO Public due to being a transitive dependency to AppImpl
+@Suppress("LongParameterList")
 public class AppConfigurationImpl constructor(
     override val appId: String,
     override val baseUrl: String = DEFAULT_BASE_URL,
+    override val encryptionKey: ByteArray?,
```

Comment:
Should we keep a copy of the encryption key at SDK level? I think that we should not keep any copy, as they should be safely stored by the user or core.

`RealmConfiguration` also stores a copy. For that case we have a core function to retrieve the key, that would help testing, but we lack of such function for the metadata encryption key.

We could address this in another PR.

Reply:
I agree, this is a bit of a smell, but I wouldn't consider it a big deal, since if you have access to read the memory where this key resides, you also have access to the open Realm and can just ask Core for the data.

Reply:
Interesting points. Currently, however, the exposed core function `realm_config_get_encryption_key` (not for metadata) is not used in any configuration class. The `actual` functions are implemented but yes, a copy is still being stored.